src/main/java/com/minecolonies/coremod/commands/GetColonyAndCitizen.java

```diff
@@ -0,0 +1,219 @@
+    private static final String NO_COLONY                   = "You haven't got a colony!";
+    private static final int    ONLY_NUMBERS_CODE           = -1;
+    private static final int    TOO_MANY_ARGUMENTS_CODE     = -2;
+    private static final int    UNKNOWN_ERROR_CODE          = -3;
+    private static final int    NOT_FOUND_CODE              = -4;
+    private static final int    NO_COLONY_CODE              = -5;
+    private static final int    SHORT_ARGUMENT_LENGTH       = 1;
+    private static final int    NORMAL_ARGUMENT_LENGTH      = 2;
+    private static final int    NAME_ARGUMENT_LENGTH        = 3;
+    private static final int    ID_AND_NAME_ARGUMENT_LENGTH = 4;
+    private static final int    TOO_MANY_ARGUMENTS_LENGTH   = 5;
+    private static final int    ARGUMENT_ZERO               = 0;
+    private static final int    ARGUMENT_ONE                = 1;
+    private static final int    ARGUMENT_TWO                = 2;
+    private static final int    ARGUMENT_THREE              = 3;
+
+    private GetColonyAndCitizen()
+    {
+        throw new IllegalAccessError("Utility Class");
+    }
+
+    /**
+     * Getting the colony ID.
+     * @param mayorID   The ID of the mayor.
+     * @param world     The world.
+     * @param args      The arguments.
+     * @return Return colony ID.
+     */
+    public static int getColonyId(@NotNull final UUID mayorID, @NotNull final World world, @NotNull final String... args)
+    {
+        int colonyId;
+        if (args.length == NORMAL_ARGUMENT_LENGTH || args.length == ID_AND_NAME_ARGUMENT_LENGTH)
+        {
+            try
+            {
+                colonyId = Integer.parseInt(args[ARGUMENT_ZERO]);
+            }
+            catch (NumberFormatException e)
+            {
+                colonyId = ONLY_NUMBERS_CODE;
+            }
+        }
+        else if (args.length == SHORT_ARGUMENT_LENGTH || args.length == NAME_ARGUMENT_LENGTH)
+        {
+            if (ColonyManager.getIColonyByOwner(world, mayorID) == null)
+            {
+                colonyId = NO_COLONY_CODE;
+            }
+            else
+            {
+                colonyId = ColonyManager.getIColonyByOwner(world, mayorID).getID();
+            }
+        }
+        else if (args.length >= TOO_MANY_ARGUMENTS_LENGTH)
+        {
+            colonyId = TOO_MANY_ARGUMENTS_CODE;
+        }
+        else
+        {
+            colonyId = UNKNOWN_ERROR_CODE;
+        }
+        if (colonyId >= 0 && ColonyManager.getColony(colonyId) == null)
+        {
+            colonyId = NOT_FOUND_CODE;
+        }
+
+
+        return colonyId;
+    }
+
+    /**
+     * Getting the citizen ID.
+     * @param colonyId          The colony ID for getting the citizen ID
+     * @param args              The given arguments
+     * @return Returns citizen ID
+     */
+    public static int getCitizenId(final int colonyId, @NotNull final String... args)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getCitizenId" is 18 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)